### PR TITLE
Adding docs about scaled_float saturation with long values

### LINKED ISSUE #105361: scale_float does not honor limits for long values
### Elasticsearch Version

7.17 / 8.x

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

N/A

### Problem Description

When using `scaled_float`, there is no range check on the value and scaling factor to be outside the `long` value range.

According to [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/number.html) for `scaled_float`:

> scaled_float - A floating point number that is backed by a long, scaled by a fixed double scaling factor

However, that assumption is not checked on document ingestion. The following mapping:

"properties": {
  "a_scaled_float": {
    "scaling_factor": 1000000,
    "type": "scaled_float"
  }
There's a scaled_float mapping configuration with value 1000000.
So, why did the number 1707339753294418 become 9223372036854.775? Shouldn't it be 1707339753294418000000?

{
  "_index": "test-index",
  "_id": "Z5t_iY0BLRoRMvLbP6e3",
  "_score": 1,
  "_source": {
    "a_scaled_float": 1707339753294418
  },
  "fields": {
    "a_scaled_float": [
      9223372036854.775
    ]
  }
}

This is due to the rounding up done [here](https://github.com/elastic/elasticsearch/blob/17bc83f481ddcfc032f6743a7a64c64000d55740/modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/ScaledFloatFieldMapper.java#L388), which produces a `long` value which is outside range, and thus returns `MAX_LONG`.

As this provides an incorrect result, and the value is outside of the `long` range, the correct behaviour should be to error on the value as other numeric types do.

### Steps to Reproduce

Set a mapping with properties:

```json
PUT test-index
{
      "mappings": {
        "properties": {
            "scaled_float": {
              "type": "scaled_float",
              "scaling_factor": 1000000
            }
        }
      }
}
```

Ingest a document:

```json
PUT test-index/_doc/1
{
  "scaled_float": 1707339753294418
}
```

This should return an error, as the number is outside the `long` range.

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by benwtrent**
closes: https://github.com/elastic/elasticsearch/issues/105361

### CODE REVIEW

File: docs/reference/mapping/types/numeric.asciidoc [hunk 1]
```diff
@@ -227,6 +227,19 @@ numeric field can't be both a time series dimension and a time series metric.
...
     This parameter is required.
 
+[[scaled-float-saturation]]
+==== `scaled_float` Saturation
+
+`scaled_float` is backed by a single `long` value. Meaning, the scaling factor
+and the value are multiplied and stored as a `long`. If the multiplication
+results in a value that is outside the range of a `long`, the value is saturated
+to the minimum or maximum value of a `long`. For example, if the scaling factor
+is +100+ and the value is +92233720368547758.08+, the expected value might be +9223372036854775808+.
```
Comment (shainaraskas) [inline docs/reference/mapping/types/numeric.asciidoc:237]: I think we can confidently say the expected value is `9223372036854775808`

```suggestion
is +100+ and the value is +92233720368547758.08+, the expected value is +9223372036854775808+.
```

File: docs/reference/mapping/types/numeric.asciidoc [hunk 2]
```diff
@@ -227,6 +227,19 @@ numeric field can't be both a time series dimension and a time series metric.
     of `scaling_factor` improve accuracy but also increase space requirements.
     This parameter is required.
 
+[[scaled-float-saturation]]
+==== `scaled_float` Saturation
```
Comment (shainaraskas) [inline docs/reference/mapping/types/numeric.asciidoc:231]:
```suggestion
==== `scaled_float` saturation
```

File: docs/reference/mapping/types/numeric.asciidoc [hunk 3]
```diff
@@ -227,6 +227,19 @@ numeric field can't be both a time series dimension and a time series metric.
...
+
+`scaled_float` is backed by a single `long` value. Meaning, the scaling factor
+and the value are multiplied and stored as a `long`. If the multiplication
```
Comment (shainaraskas) [inline docs/reference/mapping/types/numeric.asciidoc:234]:
```suggestion
`scaled_float` is stored as a single `long` value, which is the product of multiplying the original value by the scaling factor. If the multiplication
```

File: docs/reference/mapping/types/numeric.asciidoc [hunk 4]
```diff
@@ -227,6 +227,19 @@ numeric field can't be both a time series dimension and a time series metric.
...
+results in a value that is outside the range of a `long`, the value is saturated
+to the minimum or maximum value of a `long`. For example, if the scaling factor
+is +100+ and the value is +92233720368547758.08+, the expected value might be +9223372036854775808+.
+However, the value actually stored is +9223372036854775807+, the maximum value for a `long`.
```
Comment (shainaraskas) [inline docs/reference/mapping/types/numeric.asciidoc:238]:
```suggestion
However, the value that is stored is +9223372036854775807+, the maximum value for a `long`.
```

File: docs/reference/mapping/types/numeric.asciidoc [hunk 5]
```diff
@@ -227,6 +227,19 @@ numeric field can't be both a time series dimension and a time series metric.
...
+
+This can lead to unexpected results with <<query-dsl-range-query,range queries>>
+when the scaling factor is large or the value is close to the maximum value of a `long`.
```
Comment (shainaraskas) [inline docs/reference/mapping/types/numeric.asciidoc:241]: unclear whether `value` here is the scaled float or the original value before scaling. assume the former, so maybe we should say `resulting scaled value`?

**Review by shainaraskas — APPROVED** [review]

**Comment by benwtrent** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.3|[](https://github.com/elastic/elasticsearch/pull/108083)|
|✅|8.2|[](https://github.com/elastic/elasticsearch/pull/108084)|
|✅|8.1|[](https://github.com/elastic/elasticsearch/pull/108085)|
|✅|8.0|[](https://github.com/elastic/elasticsearch/pull/108086)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/108087)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

File: docs/reference/mapping/types/numeric.asciidoc (hunk 5 above)
Comment (shainaraskas) [reply docs/reference/mapping/types/numeric.asciidoc]: that makes more sense - thanks!

**Pull request merged**